src/gromacs/nblib/molecules.h

```diff
@@ -97,10 +96,10 @@ class Molecule
     template<typename T>
     Molecule& addAtom(const T& atomName, AtomType const& atomType) = delete;
 
-    void addHarmonicBond(HarmonicType harmonicBond);
+    void addHarmonicBond(const HarmonicType& harmonicBond);
```

Comment:
Same comment as for the simstate constructor applies here. Use pass by value + move to avoid the copy if called with rvalues. See below.